### PR TITLE
Separate Nova-compute and DVS to dedicated pods

### PR DESCRIPTION
**Pull request by Scsabiii**
Due to NSX-T we have to separate the Nova-compute and the DVS to dedicated pods.
This is a new pull request based on https://github.com/sapcc/helm-charts/pull/471

### CODE REVIEW

**Comment by Scsabiii** [comment]
I removed the neutron configmap hypervisors-vmware-neutron-configmap-vct.yaml

File: openstack/neutron/templates/vct-dvs-agent-configmap.yaml [hunk 1]
```diff
@@ -33,7 +34,4 @@ template: |
       vsphere_password = {= password | quote =}
 
       network_maps = "{= physical | ini_escape =}:{= bridge | ini_escape =}"
-
-      [DVS]
-      portgroup_retention_iterations = 50
```
Comment (joker-official) [inline openstack/neutron/templates/vct-dvs-agent-configmap.yaml]: You forgot to port these changes to the new file.

File: openstack/neutron/templates/vct-dvs-agent-deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+      spec:
+        containers:
+        - name: neutron-dvs-agent
+          image: {{.Values.global.imageRegistry}}/{{.Values.global.image_namespace}}/loci-neutron:{{.Values.imageVersionNeutron | required "Please set nova.imageVersionNeutron or similar" }}
+          imagePullPolicy: IfNotPresent
+          command: ["dumb-init"]
+          args: ["neutron-dvs-agent", "--config-file", "/etc/neutron/neutron.conf", "--config-file", "/etc/neutron/plugins/ml2/ml2-conf.ini", "--config-file", "/etc/neutron/plugins/ml2/ml2-vmware.ini"]
+          env:
+          - name: SENTRY_DSN
+            value: {{ .Values.neutron.sentry_dsn | quote }}
```
Comment (joker-official) [inline openstack/neutron/templates/vct-dvs-agent-deployment.yaml]: This should be changed to use the sentry operator.

File: openstack/neutron/templates/vct-dvs-agent-deployment.yaml [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+          component: agent
+          name: neutron-dvs-agent-{= name =}
+        annotations:
+          configmap-etc-hash: {{ include (print $.Template.BasePath "/configmap-etc.yaml") . | sha256sum }}
+          prometheus.io/scrape: "true"
+          prometheus.io/port: "9102"
+      spec:
+        containers:
+        - name: neutron-dvs-agent
+          image: {{.Values.global.imageRegistry}}/{{.Values.global.image_namespace}}/loci-neutron:{{.Values.imageVersionNeutron | required "Please set nova.imageVersionNeutron or similar" }}
```
Comment (joker-official) [inline openstack/neutron/templates/vct-dvs-agent-deployment.yaml]: There's not `imageVersionNeutron` in neutron.

File: openstack/neutron/templates/vct-dvs-agent-deployment.yaml (hunk 2 above)
Comment (joker-official) [reply openstack/neutron/templates/vct-dvs-agent-deployment.yaml]: Same for the nsx-t agent btw.

File: openstack/neutron/templates/vct-dvs-agent-configmap.yaml (hunk 1 above)
Comment (Scsabiii) [reply openstack/neutron/templates/vct-dvs-agent-configmap.yaml]: Thx, i missed that

**Comment by joker-official** [comment]
Tested the split in staging with a 9cbb976 rebase on master: It works, but one has to manually delete the nova-compute deployments because the vcenter-operator cannot remove a container from the deployment. Afterwards, one can deploy neutron and the dvs-agents come up as expected.

I don't want to merge this until there's a plan to roll this out, as any roll-out of nova will depend on a roll-out of neutron once this is merged. Therefore, we have to roll it through all regions in a short amount of time.

**Comment by joker-official** [comment]
Closing this because we removed DVS.

**Pull request closed without merging**